### PR TITLE
Task form page linking

### PR DESCRIPTION
**Pull request by V-X-L-U**
In the PR:
- [x] Making an API call to create a Task
- [x] Task Data Validation
- [x] Setting and Showing Error Message (code for setting error messages not complete)
- [x] Fix: Adding `DateSelector`s for `startDate` and `endDate`
- [x] ~~Getting a proper UserID (hardcoded at the moment)~~ Not needed since Chris changed the way the JWT works
- [x] ~~Getting a proper JWT (hardcoded at the moment)~~ Not needed since Chris changed the way the JWT works
- [x] (UI Fix) Fix multiline text input for "Description"

### CODE REVIEW

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (hiimchrislim) [inline client/src/pages/TaskFormPage/TaskFormPage.js]: Another check that I would also include is to make sure that the start date isn't happening before current date today

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (hiimchrislim) [inline client/src/pages/TaskFormPage/TaskFormPage.js]: We can actually omit this function since the API will accept the full entire ISOString

Comment (hiimchrislim) [reply client/src/pages/TaskFormPage/TaskFormPage.js]: Although on the documentation example it shows it without the last 5 values, it'll still accept the full entire string

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (hiimchrislim) [inline client/src/pages/TaskFormPage/TaskFormPage.js]: Since we're omitting the getISOString function (from the comment above), we can just do `(new Date()).toIsoString()` (I believe this should work)

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (hiimchrislim) [inline client/src/pages/TaskFormPage/TaskFormPage.js]: We can probably merge the updateTaskDate along with the updateTaskFormData (Reduces the amount of redundant code that we have)

Comment (hiimchrislim) [reply client/src/pages/TaskFormPage/TaskFormPage.js]: Although they're somewhat different, we can just make a small check for the dates in particular (if needed)

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (hiimchrislim) [inline client/src/pages/TaskFormPage/TaskFormPage.js]: I would change this to `if (validationResult !== "")` just because the error message doesn't seem to display (only on console)
image

Comment (hiimchrislim) [inline client/src/pages/TaskFormPage/TaskFormPage.js]: Minor comment: To also reduce the amount of if else statements, you could also do something like

```
if (validationResult !== "") {
    setErrorMessage(validationResult)
    return
}

// Other code here for the API call
```
By doing so, it reduces the amount of indentation levels and makes the code easier to read as well

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (hiimchrislim) [reply client/src/pages/TaskFormPage/TaskFormPage.js]: Actually, we can just stick with `new Date()` instead. We should only be doing a .toISOString() only when we're making the API call

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (hiimchrislim) [inline client/src/pages/TaskFormPage/TaskFormPage.js]: Referring to one of the comments on line 19, since we'll no longer have the helper function that you created nor the .toISOString method, we won't need `new Date(taskFormData.startDate)`. Rather we can just have taskFormData.startDate since it'll already be of type date.

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (hiimchrislim) [inline client/src/pages/TaskFormPage/TaskFormPage.js]: Refer to the comment on line 115 for this date selector as well

**Review by hiimchrislim — CHANGES_REQUESTED** [review]
Overall, things are pretty good! I left a few comments. If you have any questions, please let me know

**Comment by hiimchrislim** [comment]
Additionally, I think it would be good to also link up this page along with the main calendar page (i.e. After a successful create task, it should redirect to the main calendar page or if you click on the X on the create task page, it should bring you back to the calendar page as well, etc.)

Perhaps it might be better to have a "Cancel" button beside the Create button instead of the X on the top of the create task page

**Comment by hiimchrislim** [comment]
Also, as a UI fix (I missed this) but the text should probably be starting at the top rather than in the center of the description box (Unless this is the intended design)
image

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (V-X-L-U) [reply client/src/pages/TaskFormPage/TaskFormPage.js]: Added a new check as specified.

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (V-X-L-U) [reply client/src/pages/TaskFormPage/TaskFormPage.js]: Helper removed.

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (V-X-L-U) [reply client/src/pages/TaskFormPage/TaskFormPage.js]: Changed to store `Date` object instead of ISO string inside state.

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (V-X-L-U) [reply client/src/pages/TaskFormPage/TaskFormPage.js]: Added a check for "startDate" and "endDate" value to use `e` as is instead of `e.target.value`.

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (V-X-L-U) [reply client/src/pages/TaskFormPage/TaskFormPage.js]: The error message display has not been implemented (see PR checklist and standup). I've just set up a part of the error logic at the moment.

I'll finish the error display by Saturday latest. Hopefully by Friday though.

Comment (V-X-L-U) [reply client/src/pages/TaskFormPage/TaskFormPage.js]: Changed if-statement structure as suggested.

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (V-X-L-U) [reply client/src/pages/TaskFormPage/TaskFormPage.js]: Changed to just `taskFormData.startDate` since the date object is stored in the state now.

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (V-X-L-U) [reply client/src/pages/TaskFormPage/TaskFormPage.js]: Changed accordingly also.

**Comment by V-X-L-U** [comment]
> Additionally, I think it would be good to also link up this page along with the main calendar page (i.e. After a successful create task, it should redirect to the main calendar page or if you click on the X on the create task page, it should bring you back to the calendar page as well, etc.)
> 
> Perhaps it might be better to have a "Cancel" button beside the Create button instead of the X on the top of the create task page

Replaced "x" button on the top-right with a "Cancel" button to the left of the "Create" button. Routes back to `CalendarPage`. 

When a task is succesfully created, route back to `CalendarPage`.

**Comment by V-X-L-U** [comment]
> Also, as a UI fix (I missed this) but the text should probably be starting at the top rather than in the center of the description box (Unless this is the intended design) image

At the moment, I'm thinking of using `<textarea>` for this. A multiline input box was intended here. However, I will defer the implementation to the next review to prioritize linking code.

This will be done together with the `ErrorMessage` display, which is also UI-related.

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (hiimchrislim) [reply client/src/pages/TaskFormPage/TaskFormPage.js]: Noted!

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (hiimchrislim) [inline client/src/pages/TaskFormPage/TaskFormPage.js:130]: Minor comment: I would just swap the places of the Create and Cancel button (It seems more fitting to have create be on the left rather than on the right so it's flush with the rest of the Create Task Form)

Comment (hiimchrislim) [reply client/src/pages/TaskFormPage/TaskFormPage.js:130]: Minor UI Comment: Also adding some hover styles (i.e. Changing the background to be a bit darker) might also be worth considering

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (hiimchrislim) [inline client/src/pages/TaskFormPage/TaskFormPage.js]: Minor comment: This doesn't need to be a function. It can just be stored as a variable (i.e. `const navigate/routeToCalendar = navigate("/calendar")`)

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (hiimchrislim) [inline client/src/pages/TaskFormPage/TaskFormPage.js:28]: We can probably remove this set state call since we're only doing the validation check at the time the user clicks "submit/create task"

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (hiimchrislim) [reply client/src/pages/TaskFormPage/TaskFormPage.js]: Actually no need for the function, we can just use `navigate("/calendar")` directly

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (V-X-L-U) [reply client/src/pages/TaskFormPage/TaskFormPage.js:130]: Will check with the team for "Cancel" and "Create" position.

Will put the hover style improvement on hold.

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (V-X-L-U) [reply client/src/pages/TaskFormPage/TaskFormPage.js]: Directly used `navigate` instead.

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (V-X-L-U) [reply client/src/pages/TaskFormPage/TaskFormPage.js:28]: This is to prevent a user from getting stuck with an error message. If we get a validation error and make some changes to our form, the error message shouldn't persist.

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (hiimchrislim) [reply client/src/pages/TaskFormPage/TaskFormPage.js:130]: Noted!

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (hiimchrislim) [inline client/src/pages/TaskFormPage/TaskFormPage.js]: I would remove the `user_id` attribute since it's not needed (due to the changes in develop with the JWT being embedded into the cookie)

File: client/src/components/InputBox/InputBox.js
Comment (hiimchrislim) [inline client/src/components/InputBox/InputBox.js:18]: Minor design comment: Is there a way that perhaps the scroll bar could fit inside the actual box rather than overlapping it?
image

Comment (hiimchrislim) [reply client/src/components/InputBox/InputBox.js:18]: Note for future design component implementations: Generally for this Description InputBox, we should probably have a separate component for input boxes that require larger text areas but for our use case this will suffice

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (hiimchrislim) [inline client/src/pages/TaskFormPage/TaskFormPage.js:112]: For the date selector, is there any way to extend the box so it matches the length of the other boxes? (Same with the description box)
image

Comment (hiimchrislim) [reply client/src/pages/TaskFormPage/TaskFormPage.js:112]: Unless this was the intended design that we wanted

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (hiimchrislim) [inline client/src/pages/TaskFormPage/TaskFormPage.js:128]: Minor comment: I would probably add some extra margin at the bottom (so the errors are spaced out a bit better)
image

**Review by hiimchrislim — CHANGES_REQUESTED** [review]
Overall, things look good! There are just a few minor comments that I added in.
Once those are addressed/fixed, you should merge the develop branch into this one and run some final tests to ensure the functionality of the page

If you have any questions, please let me know 🙂

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (V-X-L-U) [reply client/src/pages/TaskFormPage/TaskFormPage.js:128]: Added 10px margin.

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (V-X-L-U) [reply client/src/pages/TaskFormPage/TaskFormPage.js:112]: Yep, this is the intended design. By default, the button actually expands bigger than this, but I've added a `max-width` of 120px to prevent that.

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (V-X-L-U) [reply client/src/pages/TaskFormPage/TaskFormPage.js]: Removed.

File: client/src/components/InputBox/InputBox.js
Comment (V-X-L-U) [reply client/src/components/InputBox/InputBox.js:18]: Couldn't find a proper fix for this. I could only get it to show a tiny box that is barely visible.

Comment (hiimchrislim) [reply client/src/components/InputBox/InputBox.js:18]: That's ok for our use case since we'll be focusing more on functionality than aesthetics

File: client/src/pages/TaskFormPage/TaskFormPage.js
Comment (hiimchrislim) [reply client/src/pages/TaskFormPage/TaskFormPage.js:112]: Noted

**Review by hiimchrislim — APPROVED** [review]
Looks great! Good job!

**Pull request merged**
